doc/Lorre's Block Fetching Algorithm

```diff
@@ -0,0 +1,96 @@
+Lorre's Block Fetching Algorithm:
+
+The following logic is implemented via the processBlocks function, which is a private helper of the
+more general getBlocks function.
+
+Initially, when Lorre is first run with an empty database, it connects to the head block on the chain
+at level n, and will add blocks at levels 0..n to the chain. If there is no fork at all,
+the next iteration of Lorre will run. If there are no new blocks, nothing is added to the database.
+If there are new blocks, let the head block's level be n_2. Then, blocks n+1, n+2...n_2 will be
+added to the database. Inductively, if there are never any forks, then at iteration i of Lorre,
+Lorre will add blocks n_{i-1} + 1, n_{i-1} + 2...n_i to the database.
+
+ASSUMPTION: In a non-forked chain, there can only be one block at each level. [i]
+
+------------------------------------------------------------------------------------------------------
+
+Lorre's algorithm for fetching blocks that occur as a result of a fork:
+
+The following logic is implemented via the followFork function, which is a private helper of the
+more general getBlocks function.
+
+In the normal case, we place blocks into a blocks table, with the fields being all of the associated
+metadata. In the forked case, we will have an invalidated blocks table to invalidate blocks that are
+not internally consistent with the new chain being recognized by Lorre. Here, the necessary fields are
+block_id (the block's hash), level, and an is_invalidated boolean flag.
+
+ASSUMPTION: Internal consistency with our database is all we can hope for, as there is no notion of a
+globally correct chain. [ii]
+
+Suppose a fork happens, and an iteration of Lorre collects a batch of blocks with levels m, m+1...n.
+Blocks at levels m to n are added to the blocks table. we will then consider m's predecessors, as those
+are the blocks that may conflict with our internal view of what the current chain looks like.
+
+For the block at level m-1, we perform two database checks: Is this block currently in the Blocks Table,
+and is this block currently in the Invalidated Blocks Table?
+
+Case 1: Block at level m-1 is in the blocks table, but is not in the invalidated blocks table.
```

Comment:
just a trivial thought, does the different Cases (1,2,3,4) have any easy-to-remind name to talk about them? e.g.
 - case 1 unforked block
 - case 2 forked substitute
 - case 3 restored block
 - case 4 inconsistent

Reply:
I'll look into this! I'll need to take care with proper, non misleading naming